internal/terraform/module/module_ops_test.go

```diff
@@ -931,6 +942,141 @@ func TestPreloadEmbeddedSchema_raceCondition(t *testing.T) {
+	}
+
+	// ignore job state
+	ctx = job.WithIgnoreState(ctx, true)
+
+	// say we're coming from did_change request
+	fooURI, _ := filepath.Abs(filepath.Join(singleFileModulePath, "foo.tf"))
+	x := lsctx.RPCContextData{
+		Method: "textDocument/didChange",
+		URI:    uri.FromPath(fooURI),
+	}
+	ctx = lsctx.WithLanguageId(ctx, ilsp.Terraform.String())
+	ctx = lsctx.WithRPCContext(ctx, x)
+	err = ParseModuleConfiguration(ctx, testFs, ss.Modules, singleFileModulePath)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	after, err := ss.Modules.ModuleByPath(singleFileModulePath)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// test if foo.tf is not the same as first seen
+	if before.ParsedModuleFiles["foo.tf"] == after.ParsedModuleFiles["foo.tf"] {
+		t.Fatal("file should mismatch")
+	}
+
+	// test if main.tf is the same as first seen
+	if before.ParsedModuleFiles["main.tf"] != after.ParsedModuleFiles["main.tf"] {
+		t.Fatal("file mismatch")
+	}
+
+	// examine diags should change for foo.tf
+	if before.ModuleDiagnostics["foo.tf"][0] == after.ModuleDiagnostics["foo.tf"][0] {
+		t.Fatal("diags should mismatch")
+	}
+
+	// examine diags should change for main.tf
+	if before.ModuleDiagnostics["main.tf"][0] != after.ModuleDiagnostics["main.tf"][0] {
+		t.Fatal("diags should match")
+	}
+}
+func TestParseModuleConfiguration_ignore_tfvars(t *testing.T) {
+	ctx := context.Background()
+	ss, err := state.NewStateStore()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	testData, err := filepath.Abs("testdata")
+	if err != nil {
+		t.Fatal(err)
+	}
+	testFs := filesystem.NewFilesystem(ss.DocumentStore)
+
+	singleFileModulePath := filepath.Join(testData, "single-file-change-module")
+
+	err = ss.Modules.Add(singleFileModulePath)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	ctx = lsctx.WithRPCContext(ctx, lsctx.RPCContextData{})
+	err = ParseModuleConfiguration(ctx, testFs, ss.Modules, singleFileModulePath)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	before, err := ss.Modules.ModuleByPath(singleFileModulePath)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// ignore job state
+	ctx = job.WithIgnoreState(ctx, true)
+
+	// say we're coming from did_change request
+	fooURI, _ := filepath.Abs(filepath.Join(singleFileModulePath, "example.tfvars"))
+	x := lsctx.RPCContextData{
+		Method: "textDocument/didChange",
+		URI:    uri.FromPath(fooURI),
+	}
+	ctx = lsctx.WithLanguageId(ctx, ilsp.Tfvars.String())
+	ctx = lsctx.WithRPCContext(ctx, x)
+	err = ParseModuleConfiguration(ctx, testFs, ss.Modules, singleFileModulePath)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	after, err := ss.Modules.ModuleByPath(singleFileModulePath)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// example.tfvars should be the same
+	if before.ParsedModuleFiles["example.tfvars"] != after.ParsedModuleFiles["example.tfvars"] {
+		t.Fatal("file should match")
+	}
+
+	// diags should not change for example.tfvars
+	if before.ModuleDiagnostics["example.tfvars"] != nil {
+		t.Fatal("there should be no diags for tfvars files right now")
+	}
```

Comment:
```suggestion
	// example.tfvars should be missing
	_, beforeExists := before.ParsedModuleFiles["example.tfvars"]
	if beforeExists {
		t.Fatal("example.tfvars file should be missing")
	}
	_, afterExists := after.ParsedModuleFiles["example.tfvars"]
	if afterExists {
		t.Fatal("example.tfvars file should be missing")
	}

	// diags should be missing for example.tfvars
	if _, ok := before.ModuleDiagnostics["example.tfvars"]; ok {
		t.Fatal("there should be no diags for tfvars files right now")
	}
```
I'm kind of surprised this isn't crashing, because there is no entry for `example.tfvars` in that map, but I suppose we just end up comparing two `nil`s.

Regardless though I think that map entry existence is what we should be testing for.